src/Symfony/Component/DependencyInjection/Loader/FileLoader.php

```diff
@@ -155,14 +156,33 @@ public function registerClasses(Definition $prototype, string $namespace, string
                     continue;
                 }
                 if ($this->env) {
-                    $attribute = null;
-                    foreach ($r->getAttributes(When::class, \ReflectionAttribute::IS_INSTANCEOF) as $attribute) {
+                    $excluded = true;
+                    $whenAttributes = $r->getAttributes(When::class, \ReflectionAttribute::IS_INSTANCEOF);
+                    $notWhenAttributes = $r->getAttributes(WhenNot::class, \ReflectionAttribute::IS_INSTANCEOF);
+
+                    if ($whenAttributes && $notWhenAttributes) {
+                        throw new LogicException(sprintf('The "%s" class cannot have both #[When] and #[WhenNot] attributes.', $class));
+                    }
+
+                    if ([] === $whenAttributes && [] === $notWhenAttributes) {
```

Comment:
such typed comparisons are needlessly noisy ;)
```suggestion
                    if ([] === $whenAttributes && [] === $notWhenAttributes) {
```

Reply:
Replaced by `empty()` calls, I guess this is what you meant?

Reply:
empty is worse actually :D 
`if (!$whenAttributes && !$notWhenAttributes) {`
and same in all similar places

Reply:
Ah 😄 Right, it should be better now